server/controllers/personal/personalController.ts

```diff
@@ -379,22 +388,62 @@ export default class PersonalController {
     }
   }
 
-  textInput(fieldData: TextFieldData) {
-    const { pageTitle, hintText, auditPage, fieldName, url, inputClasses } = fieldData
+  cityOrPlaceOfBirthTextInput = (fieldData: TextFieldData) =>
+    this.textInput(fieldData, this.getCityOrPlaceOfBirth.bind(this), this.setCityOrPlaceOfBirth.bind(this))
+
+  private async getCityOrPlaceOfBirth(req: Request): Promise<string> {
+    return Promise.resolve(req.middleware?.inmateDetail?.birthPlace)
```

Comment:
I suspect the Promise.resolve isn't necessary. Just having it as a async function should automatically wrap in promise so you should just be able to return the value